### PR TITLE
Reduce dependency on Lodash and use ES6 syntax

### PR DESCRIPTION
**Pull request by ImedAdel**
## Description

Removed `_.each`, `_.get`, and `_.uniq`.

### CODE REVIEW

File: docs/docs/adding-tags-and-categories-to-blog-posts.md [hunk 1]
```diff
@@ -211,18 +211,18 @@ exports.createPages = ({ actions, graphql }) => {
     // Tag pages:
     let tags = []
     // Iterate through each post, putting all found tags into `tags`
-    _.each(posts, edge => {
-      if (_.get(edge, "node.frontmatter.tags")) {
-        tags = tags.concat(edge.node.frontmatter.tags)
-      }
-    })
+    posts.forEach(edge => {
+			if (edge.node.frontmatter.tags) {
```
Comment (poyea) [inline docs/docs/adding-tags-and-categories-to-blog-posts.md]: I'm guessing this line fails in CircleCI build.

Comment (lannonbr) [reply docs/docs/adding-tags-and-categories-to-blog-posts.md]: Resolved the linting errors

File: docs/docs/adding-tags-and-categories-to-blog-posts.md [hunk 2]
```diff
@@ -211,18 +211,18 @@ exports.createPages = ({ actions, graphql }) => {
     // Tag pages:
     let tags = []
     // Iterate through each post, putting all found tags into `tags`
-    _.each(posts, edge => {
-      if (_.get(edge, "node.frontmatter.tags")) {
+    posts.forEach(edge => {
+      if (edge.node.frontmatter.tags) {
```
Comment (pieh) [inline docs/docs/adding-tags-and-categories-to-blog-posts.md]: `_.get` was useful here because `frontmatter` can be `null` (not only `tags`)

**Comment by DSchau** [comment]
@ImedAdel thanks for your contribution, here!

Unfortunately, we're not going to merge this at this time. As @pieh noted, we're not functionally equivalent to the previous implementation which is something we'd like to maintain.

Also, this is Node.js code, so while we always want to focus on thin and light dependencies, lodash is a pretty reasonable one to include and the functionality is well-tested, robust, and oftentimes more performant than a similar hand-rolled implementation.

From this basis, we're going to close this at this time.

If it's helpful, definitely take a look at [our open issues][open-issues] for ideas for a future PR, and please reach out to us on Twitter at @gatsbyjs with questions.

Thanks again, and we look forward to seeing more PRs from you in the future! :muscle: :purple_heart:

[open-issues]: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

**Pull request closed without merging**